server/src/main/java/org/opensearch/index/translog/transfer/FileTransferTracker.java

```diff
@@ -41,6 +41,10 @@ public void onSuccess(TransferFileSnapshot fileSnapshot) {
         });
     }
 
+    public void add(String file) {
+        fileTransferTracker.put(file, TransferState.SUCCESS);
```

Comment:
Can this safely override previous entry for `file`, if exists?

Reply:
Yes. Transition to `TransferState.SUCCESS` is allowed for a give `file`.

Reply:
Did the change with safety checks.